lib/matplotlib/backends/backend_agg.py

```diff
@@ -527,7 +563,14 @@ def print_png(self, filename_or_obj, *args, **kwargs):
         else:
             close = False
         try:
-            _png.write_png(renderer._renderer, filename_or_obj, self.figure.dpi)
+            if kwargs.get('flatten', False):
+                w, h = int(renderer.width), int(renderer.height)
+                img = renderer._renderer.buffer_rgba()
+                img = np.array(memoryview(img)).reshape((h,w,4))
+                img = self.flatten_rgba(img)
+                _png.write_png(img, filename_or_obj, self.figure.dpi)
+            else:
+                _png.write_png(renderer._renderer, filename_or_obj, self.figure.dpi)
```

Comment:
No -- it shouldn't matter.  _write_png just converts whatever is passed in to a numpy array.